experimental/auth-logic/wrappers/provenance_wrapper.go

```diff
@@ -50,3 +57,19 @@ func (p provenanceWrapper) EmitStatement() (UnattributedStatement, error) {
 		Contents: fmt.Sprintf(`expected_hash("%s::Binary", sha256:%s).`, applicationName,
 			expectedHash)}, nil
 }
+
+// GetAppNameFromProvenance parses a provenance file and emits the name of the
+// application it is about. This is useful for generating principal names,
+// for example.
+func GetAppNameFromProvenance(provenanceFilePath string) (string, error) {
+	provenance, provenanceErr := slsa.ParseProvenanceFile(provenanceFilePath)
+	if provenanceErr != nil {
+		return "", provenanceErr
+	}
+	
+  if len(provenance.Subject) < 1 {
```

Comment:
I think this should confirm there is exactly 1 subject. Currently it will accept any number of subjects, but will only select the first one.

Reply:
OK. I didn't know that was how provenance files were supposed to work. Why is the subject field an array if it is always of size exactly 1?

fixed anyway